docs/en/operations/settings/settings.md

```diff
@@ -4653,6 +4653,40 @@ Result:
 └─────────────┴───────────┘
 ```
 
+## script_line_number {#script_line_number}
+Specifies the value for the `script_line_number` field of the [system.query_log](../system-tables/query_log.md) table.
+It improves readability of logs, and adds additional information, by indicating in which line a query is located.
+Default value is 0.
+**Example**
+
+Query:
+
+``` sql
+DROP DATABASE IF EXISTS db;
+DROP DATABASE IF EXISTS db1;
+DROP DATABASE IF EXISTS db2;
+
+CREATE DATABASE db;
+CREATE DATABASE db1;
+CREATE DATABASE db2;
+
+SYSTEM FLUSH LOGS;
+SELECT query, script_line_number FROM system.query_log WHERE type=1 AND log_comment = 'script_line_number_test' limit 6;
+```
+
+Result:
+
+``` text
+┌─query────────────────────────┬─script_line_number─┐
+│ DROP DATABASE IF EXISTS db;  │                  1 │
+│ DROP DATABASE IF EXISTS db1; │                  2 │
+│ DROP DATABASE IF EXISTS db2; │                  3 │
```

Comment:
What happened to script_line_number = 4?

Reply:
![image](https://github.com/user-attachments/assets/880ad844-c11b-422a-9a41-ea6cfc195fa5)
It's empty, no query here, so it won't be displayed